dev/how-to/troubleshoot/tidb-in-kubernetes.md

```diff
@@ -18,3 +56,206 @@ TiDB Operator 使用 PV (Persistent Volume)/PVC (Persistent Volume Claim) 来存
 ```shell
 helm install pingcap/tidb-cluster -n ${releaseName} --namespace=${namespace} --version=v1.0.0-beta.3 -f values.yaml
 ```
+
+## Pod 未正常创建
+
+通过 `helm install` 创建集群后，如果 Pod 没有创建，则可以通过以下方式进行诊断：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl get tidbclusters -n ${namespace}
+kubectl get statefulsets -n ${namespace}
+kubectl describe statefulsets -n ${namespace} ${cluster-name}-pd
+```
+
+## Pod 之间网络不通
+
+针对 TiDB 集群而言，绝大部分 Pod 间的访问均通过 Pod 的域名（使用 Headless Service 分配）进行，例外的情况是 TiDB Operator 在收集集群信息或下发控制指令时，会通过 PD Service 的 `ServiceName` 访问 PD 集群。
+
+当通过日志或监控确认 Pod 间存在网络连通性问题，或根据故障情况推断出 Pod 间网络连接可能不正常时，可以按照下面的流程进行诊断，逐步缩小问题范围：
+
+1. 确认 Service 和 Headless Service 的 Endpoints 是否正常：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl -n ${namespace} get endpoints ${cluster_name}-pd
+    kubectl -n ${namespace} get endpoints ${cluster_name}-tidb
+    kubectl -n ${namespace} get endpoints ${cluster_name}-pd-peer
+    kubectl -n ${namespace} get endpoints ${cluster_name}-tikv-peer
+    kubectl -n ${namespace} get endpoints ${cluster_name}-tidb-peer
+    ```
+
+    以上命令展示的 `ENDPOINTS` 字段中，应当是由逗号分隔的 `cluster_ip:port` 列表。假如字段为空或不正确，请检查 Pod 的健康状态以及 `kube-controller-manager` 是否正常工作。
+
+2. 进入 Pod 的 Network Namespace 诊断网络问题：
+
+    {{< copyable "shell-regular" >}}
+
+    ```
+    tkctl debug -n ${namespace} ${pod_name}
+    ```
+
+    远端 shell 启动后，使用 `dig` 命令诊断 DNS 解析，假如 DNS 解析异常，请参照 [诊断 Kubernetes DNS 解析](https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/) 进行故障排除：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    dig ${HOSTNAME}
+    ```
+
+    使用 `ping` 命令诊断到目的 IP 的三层网络是否连通（目的 IP 为使用 `dig` 解析出的 ClusterIP）:
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    ping ${TARGET_IP}
+    ```
+
+    假如 ping 检查失败，请参照[诊断 Kubernetes 网络](https://www.praqma.com/stories/debugging-kubernetes-networking/)进行故障排除。
+
+    假如 ping 检查正常，继续使用 `telnet` 检查目标端口是否打开：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    telnet ${TARGET_IP} ${TARGET_PORT}
+    ```
+
+    假如 `telnet` 检查失败，则需要验证 Pod 的对应端口是否正确暴露以及应用的端口是否配置正确：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    # 检查端口是否一致
+    kubectl -n ${namespace} get po ${pod_name} -ojson | jq '.spec.containers[].ports[].containerPort'
+
+    # 检查应用是否被正确配置服务于指定端口上
+    # PD
+    kubectl -n ${namespace} -it exec ${pod_name} -- cat /etc/pd/pd.toml | grep client-urls
```

Comment:
If users just config the customized changes in values.yaml, the `grep` will return null.